### PR TITLE
[border-router] send RA message when RA parameters are updated

### LINKED ISSUE #6563: [github-actions] test_radvd_coexist.py fails intermittently
[logs_73915.zip](https://github.com/openthread/openthread/files/6424353/logs_73915.zip)
[thread-border-router-results.zip](https://github.com/openthread/openthread/files/6424354/thread-border-router-results.zip)

### PR DESCRIPTION
**Pull request by wgtdkp**
Resolves #6563.

### CODE REVIEW

File: src/core/border_router/routing_manager.cpp [hunk 1]
```diff
@@ -1198,6 +1189,27 @@ bool RoutingManager::NetworkDataContainsOmrPrefix(const Ip6::Prefix &aPrefix) co
     return contain;
 }
 
+// Update the `mRouterAdvMessage` with given Router Advertisement message.
+// Returns a boolean which indicates whether there are changes of `mRouterAdvMessage`.
+bool RoutingManager::UpdateRouterAdvMessage(const RouterAdv::RouterAdvMessage &aRouterAdvMessage)
+{
+    RouterAdv::RouterAdvMessage oldRouterAdvMessage;
```
Comment (simonlingoogle) [inline src/core/border_router/routing_manager.cpp:1196]:
```suggestion
    bool changed = (aRouterAdvMessage != mRouterAdvMessage);
    VerifyOrExit(changed);
```

Save the copying `oldRouterAdvMessage = mRouterAdvMessage` ?

File: src/core/border_router/routing_manager.cpp [hunk 2]
```diff
@@ -959,16 +959,7 @@ void RoutingManager::HandleRouterAdvertisement(const Ip6::Address &aSrcAddress,
...
-        {
-            mRouterAdvMessage.SetToDefault();
-        }
-        else
-        {
-            mRouterAdvMessage = *routerAdvMessage;
-            // TODO: add a timer for invalidating the learned RA parameters
-            // for cases that the other RA daemon crashed or is force killed.
-        }
+        needReevaluate = UpdateRouterAdvMessage(*routerAdvMessage);
```
Comment (simonlingoogle) [inline src/core/border_router/routing_manager.cpp:962]:
```suggestion
        needReevaluate |= UpdateRouterAdvMessage(*routerAdvMessage);
```

File: src/core/border_router/routing_manager.cpp (hunk 1 above)
Comment (wgtdkp) [reply src/core/border_router/routing_manager.cpp:1196]: Please notice that we don't always set `mRouterAdvMessage` to `aRouterAdvMessage`: we set `mRouterAdveMessage` to default values when `aRouterAdvMessage` has zero Router Lifetime.

Comment (simonlingoogle) [reply src/core/border_router/routing_manager.cpp:1196]: SGTM

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**